include/vcpkg/portfileprovider.h

```diff
@@ -69,4 +69,8 @@ namespace vcpkg::PortFileProvider
     std::unique_ptr<IVersionedPortfileProvider> make_versioned_portfile_provider(const vcpkg::VcpkgPaths& paths);
     std::unique_ptr<IOverlayProvider> make_overlay_provider(const vcpkg::VcpkgPaths& paths,
                                                             View<std::string> overlay_ports);
+    std::unique_ptr<IOverlayProvider> make_manifest_provider(const Path& manifest_path,
+                                                             std::unique_ptr<SourceControlFile>&& manifest_scf);
+    std::unique_ptr<IOverlayProvider> make_combined_overlay_provider(
+        std::vector<std::unique_ptr<IOverlayProvider>>&& providers);
```

Comment:
```suggestion
    std::unique_ptr<IOverlayProvider> make_manifest_provider(const Path& manifest_path,
                                                             std::unique_ptr<SourceControlFile>&& manifest_scf, View<std::string> overlay_paths);
```

While normally I'm all for making "compositions" like this, I think in this case the extensibility isn't warranted. The manifest must _always_ come first, you can only have one, and the rest of the overlay paths are already "composable" via concatenation.

Reply:
I used it here to avoid needing to form the concatenated vector, but I'm OK with that change too :)

https://github.com/microsoft/vcpkg-tool/pull/582/files#diff-f48df50603b746299fa327ec03ff0efba0f44c1ce81a5c892b6737374c418cd9R1110

Will fix.

Reply:
Done!